### ISSUE TITLE
Utilize Facebook Insight Services for Reaction Feature Extraction

### ISSUE THREAD

**Issue by erikschlegel** [CatalystCode/project-fortis#21]

**Comment by c-w** [comment]
## Investigation on Facebook Insights API ##

Extracts from the [docs](https://developers.facebook.com/docs/graph-api/reference/v2.9/insights), emphasis mine.

### Permissions ###

**Any valid access token can be used for publicly available metrics (currently page_fans_country or page_storytellers_by_country).**
A user access token with read_insights permission can retrieve metrics for all pages owned by this user.
A page access token with read_insights permission can retrieve all metrics for that Page.

## page_storytellers_by_country ##
The number of People Talking About the Page by user country.

The data for page_storytellers_by_country contains data for the top 45 countries overall, not the top 45 countries for your page. This means:

**If your page has no data for the overall top 45 countries, they will not appear in the list
If you have users that are not from the top 45 countries, they will not appear in the list**
Therefore the sum of all breakdwon values returned by page_storytellers_by_country may not be the same value returned by page_storytellers.

### page_fans_country ###
Aggregated Facebook location data, sorted by country, about the **people who like your Page**.

## Summary ##

**tl;dr**: It doesn't seem as though the Insights API is useful to us.

We can only get a few insights for pages that we don't own:
(1) number of likes by country
(2) number of mentions by country for the top 45 countries.

Metric (1) is not interesting for Fortis. Metric (2) is more interesting but severely limited because it only gives us data for the top-45 countries in terms of mentions globally, i.e., they'll likely be US, Canada, UK, etc. so very likely there will never be data for countries relevant to Fortis (e.g. Somalia).

**Comment by c-w** [comment]
Reviewed with Erik. This won't be of use to us. Resolving.
